test/tests/functional/pbs_job_array.py

```diff
@@ -659,3 +659,25 @@ def test_subjob_stdfile_custom_dir(self):
                 if f_name not in file_list:
                     raise self.failureException("std file " + f_name
                                                 + " not found")
+
+    def test_subjob_wrong_state(self):
+        """
+        Test that after submitting a job and restarting the server,
+        the subjobs are not in the wrong substate and can be scheduled
+        """
+        a = {'resources_available.ncpus': 500}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        a = {ATTR_J: '1-500'}
+        j = Job(attrs=a)
+        jid = self.server.submit(j)
+        self.server.restart()
+        sjids = [j.create_subjob_id(jid, x) for x in range(1, 501)]
+        for sjid in sjids:
+            state = self.server.status(JOB, id=sjid)[0]['job_state']
+            if state == 'Q':
```

Comment:
Do status(JOB) instead of doing a status() per subjob.  This will get you all the subjobs in one go.